llvm/lib/TableGen/JSONBackend.cpp

```diff
@@ -153,38 +147,36 @@ void JSONEmitter::run(raw_ostream &OS) {
     obj["!fields"] = std::move(fields);
 
     json::Array superclasses;
-    for (const auto &SuperPair : Def.getSuperClasses())
-      superclasses.push_back(SuperPair.first->getNameInitAsString());
+    // Add this def to the instance list for each of its superclasses.
+    for (const auto &[SuperClass, Loc] : Def->getSuperClasses()) {
+      std::string SuperName = SuperClass->getNameInitAsString();
+      superclasses.push_back(SuperName);
+      instance_lists[SuperName].push_back(Name);
+    }
+
     obj["!superclasses"] = std::move(superclasses);
 
     obj["!name"] = Name;
-    obj["!anonymous"] = Def.isAnonymous();
+    obj["!anonymous"] = Def->isAnonymous();
 
     json::Array locs;
-    for (const SMLoc Loc : Def.getLoc())
+    for (const SMLoc Loc : Def->getLoc())
       locs.push_back(SrcMgr.getFormattedLocationNoOffset(Loc));
     obj["!locs"] = std::move(locs);
 
     root[Name] = std::move(obj);
-
-    // Add this def to the instance list for each of its superclasses.
-    for (const auto &SuperPair : Def.getSuperClasses()) {
-      auto SuperName = SuperPair.first->getNameInitAsString();
-      instance_lists[SuperName].push_back(Name);
-    }
   }
 
   // Make a JSON object from the std::map of instance lists.
   json::Object instanceof;
-  for (auto kv: instance_lists)
-    instanceof[kv.first] = std::move(kv.second);
+  for (auto &[ClassName, Instances] : instance_lists)
+    instanceof [ ClassName ] = std::move(Instances);
```

Comment:
Is this clang-formatted? Looks odd.

Reply:
`instanceof` is a JS/Java keyword. It seems this issue has been fixed in April this year: https://github.com/llvm/llvm-project/pull/88085

CI was happy about this, so I am suspecting that it has not picked up the newest clang-format (Assuming CI uses a clang-format that was installed and not built from LLVM sources ToT).

Reply:
yeah. CI for "Check code formatting" installs clangformat 18.1.7. So I guess when we upgrade it to a version that has the fix, we will have to update the formatting here.

Reply:
Resolving.